clients/admin-ui/src/features/common/Restrict.tsx

```diff
@@ -0,0 +1,31 @@
+import { useAppSelector } from "~/app/hooks";
+import { selectThisUsersScopes } from "~/features/user-management";
+import { ScopeRegistry } from "~/types/api";
+
+export const useHasPermission = (scopes: ScopeRegistry[]) => {
+  const userScopes = useAppSelector(selectThisUsersScopes);
+  return (
+    userScopes.filter((userScope) => scopes.includes(userScope)).length > 0
+  );
+};
+
+const Restrict = ({
```

Comment:
this component is pretty basic right now in just deciding whether or not to render something based on if the user has a scope. we'll need to extend it once the data steward flow is in place